src/components/AppToolbarCentered.vue

```diff
@@ -51,31 +52,69 @@ export default {
     showBack: {
       type: Boolean,
       default: true
+    },
+    noMaxWidth: {
+      type: Boolean,
+      default: false
+    },
+    sticky: {
+      type: Boolean,
+      default: false
     }
   },
   setup() {
+    const route = useRoute()
+    const router = useRouter()
+
+    const className = 'app-toolbar-centered'
+
+    const goBack = () => {
+      if (route.query?.from?.includes('chats')) {
+        router.push(route.query?.from)
+        return
+      }
+
+      if (route.query?.fromChat) {
+        router.back()
+        return
+      }
+
+      const parentRoute = route.matched.length > 1 ? route.matched[route.matched.length - 2] : null
```

Comment:
Consider using a modern approach ([Array.prototype.at](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/at)) to access an element from array.